### DISCUSSION TITLE
Semester 4 Week 7

### DISCUSSION THREAD

**Discussion by Stev3eM** [SE-TINF22B6/CookHub#244] [Show and tell]
Hello backpackers,

Bla bla bla... new week... check it out:

<br>

## Frontend

✅ Fixed some automated redirections from specific pages

✅ NavBar is now responsive and switches to mobile view when width < 1100px

| `Normal view` | `Mobile view` |
| --- | --- |
| <img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/06342e8b-c948-4fc0-b744-254adec3a602" width="600" /> | <img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/d6459a77-a353-48e0-ba0f-4e801af3e24a" height="400" /> |

<br>

✅ Reworked several pages to be responsive for PC, Tablet and Mobile:
- Landing page
- Login page
- Impressum
- AboutUs

<img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/c76b286f-6e5b-4353-81d2-84b00120ebdf" height="400" /> <img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/0af6d0e8-8217-410f-8a87-26da290b1892" height="400" /> <img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/2e9031ab-cc2f-4415-97b0-5afdd713c1c3" height="400" />

<br>

✅ Adjustments to Carlos_PopUp helper button 
- small changes to button design
- when inserting the button you can now specify the buttonTextColor, buttonBorderColor and backgroundColor via parameter due to different background colors within the pages where it's going to get implemented

<br>

## Backend

✅ Fixed a bug which prevented the user to log out

✅ Fixed a bug where adventure texts could not be saved in the database
 - some texts were too long for the domain constraints, which is now fixed by using a variable-length string data type

## Clean Code

> We spend some time checking the code to clean it up. Therefore we used the following principles:

- Avoid Hard-Coded Numbers. (except of constants)
- Use Meaningful and Descriptive Names. (also cleaning up html & css with additional naming conventions)
- Use Comments Sparingly, and When You Do, Make Them Meaningful. (always)
- Write Short Functions That Only Do One Thing. (still have a couple of bigger functions which could be cleaned in the end)
- Follow the DRY (Don't Repeat Yourself) Principle and Avoid Duplicating Code or Logic.
- Follow Established Code-Writing Standards. (e.g. solid principles)

<br>

## Testing

- no slenium tests yet
- no unit tests in frontend yet

<br>

## Team Building

<img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/d97884d6-fb6f-4fad-94b7-28ebae871978" height="250" /> <img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/951f10d6-7958-44a5-ab6b-76d51dba346a" height="250" />
<img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/3e484f65-1f2f-47e5-8b37-c20ff3af63a8" height="250" />

<br><br>

See you next time!

**Your**   <img width="50" alt="image" src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/1a218c37-d3b0-4c88-9ae1-827b5a4eba42">   **Team**

**Comment by Nahlam4** [comment]
Hey Cookhub Team,

Looks like you are getting close to your final product. It is nice that you are working on making your product accessible from different devices. Also I like your measures to ensure Clean Code. 
Good Progress and keep it up 👍🏼 - DHBWhub Team

**Comment by Siimonhk** [comment]
Hey There Cookhub,

We are impressed with your overall progress so far, We like the inclusion of having the accessibility to use it on different platforms. Seems like yall are near completion. Keep up the good work. - Cookhub
